docs/amber.md

```diff
@@ -0,0 +1,138 @@
+generated pipeline. We generate names for all of the buffers in the file. The
+framebuffer is named `framebuffer`. The generated depth buffer is named
+`depth_buffer`. For each buffer seen we generate a name of `AutoBuf-<num>` where
+the number is the current number of buffers seen counting from 0.
+
+The VkScript parser is broken into three major chunks. The `Parser`,
+`SectionParser` and `CommandParser`. The `Parser` is the overall driver for the
+parser. The `SectionParser` breaks the input file into the overall chunks (each
+of the sections separated by the \[blocks]). The `CommandParser` parses the
+`[test]` section specifically. For other sections they're parsed directly in
+the `Parser` object.
+
+### Parsed Representation
+The `Script` object is the owner for all of our data. Other objects hold
+pointers but the script holds the `unique_ptr` for parsed objects.
+
+```
+                                        +--------+                 +--------------+
+                                        | Script |---------------->| Requirements |
+                                        +--------+                 +--------------+
+                                             |
+             +------------------+------------+--------+----------------------+
+             |                  |                     |                      |
+             v                  v                     v                      v
+      +---------+          +---------+        +---------+            +---------+
+      |  +--------+        |  +---------+     |  +----------+        |  +---------+
+      +--|  +--------+     +--|  +--------+   +--|  +----------+     +--|  +---------+
+         +--| Shader |        +--| Buffer |      +--| Pipeline |        +--| Command |
+            +--------+           +--------+         +----------+           +---------+
+               ^                  ^  ^                 |  |  ^               |    |
+               |                  |  |                 |  |  +---------------+    |
+Entry point    |                  |  +-----------------+  |                       |
+Optimizations  |                  |  Descriptor Set       |                       |
+Type           |                  |  Binding              |                       |
+Compiled Shader|                  |  Attachment Location  |                       |
+               |                  |                       |                       |
+               +------------------------------------------+                       |
+                                  |                                               |
+                                  +-----------------------------------------------+
+                                                        BufferCommand
+```
+
+A `Script` is made up for shaders, pipelines, buffers and commands. Pipelines
+contain shaders and buffers. The buffers hold both image and storage buffers.
```

Comment:
Instead of "The buffers hold both image and storage buffers", suggest:  "An Amber buffer corresponds to either a buffer resource or an image resource in the backend engine's API. "

(Except that sentence only belongs in one place in this doc, and I already suggested it above.)

Reply:
added here